frontend/src/components/housing/submission/SubmissionsNavigator.vue

```diff
@@ -161,6 +164,27 @@ function assignEnquiries() {
   });
 }
 
+/* eslint-disable max-len */
+// set multiPermitsNeeded property of each submission to Yes/No (count) if the submission have more than one permit with needed Yes
+function assignMultiPermitsNeeded() {
+  submissions.value.forEach((sub) => {
+    let multiPermitsNeededCount = 0;
+    const submissionPermits = permits.value.filter((p) => p.activityId === sub.activityId);
+    if (submissionPermits) {
+      submissionPermits.forEach((permit) => {
+        if (permit.needed?.toUpperCase() === BasicResponse.YES.toUpperCase()) {
+          multiPermitsNeededCount++;
+        }
+      });
```

Comment:
Instead of iterating through the list of permits to filter for the permits relevant to this submission on line 172, then iterating through that list again I suggest increasing the `multiPermitsNeededCount` during the first pass. That might prevent needing the second iteration through the list of permits.